### DISCUSSION TITLE
Not able to mute in latest version

### DISCUSSION THREAD

**Discussion by carefulcomputer** [MonitorControl/MonitorControl#1275] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

I have a Dell U3419W and use monitor control. Recently I updated to latest version. In previous versions I could reduce the volume to 0 and sound would cut off. However , in latest version I can control the volume but it doesn't mute if I reduce the slider to 0. I have tried enabling 'Enable Mute DDC Command' however monitor still doesn't mute.

### Expected behavior

When slider goes to 0, volume should mute.

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.6
- Mac model: Macbook Pro
- MonitorControl version: 4.0.2 build 6965
- Monitor(s): Dell U3419W
- Monitor Cable(s)/Connection(s): 
- Apple Silicon/M1 (yes or no):no
```

**Comment by waydabber** [comment]
Which version did you use previously that worked?

**Comment by carefulcomputer** [comment]
sorry i do not remember but I assume it was from last few months . I will try rolling back and reproduce the issue.

**Comment by waydabber** [comment]
I see. Well, nothing changed in this regard from 4.0.0 or 4.0.1 to 4.0.2 so I actually don't know where the problem lies. :)

**Comment by carefulcomputer** [comment]
That was it ! I rolled back to 4.0.0 and able to mute by moving slider to 0 ! So something changed between 4.0.0 (build 6851) and later versions.

**Comment by waydabber** [comment]
what about 4.0.1? I don't really remember changing anything regarding this. But I might remember wrong. :)

**Comment by carefulcomputer** [comment]
just tried 4.0.1. That has very weird behavior. It would work sometimes and sometimes not (i.e. moving slider to 0 position would mute sound sometimes but not other times). Could it somehow be linked to dragging speed/over drag ?

**Comment by carefulcomputer** [comment]
I can consistently mute in 4.0.0. Intermittently (within a gap of few seconds) in 4.0.1 and almost never in 4.0.2

**Comment by waydabber** [comment]
Well, there was some changes related to how the sliders and DDC works - redundant DDC writes were mostly eliminated by using a debouncing algorithm. It might be the display does not respond to some DDC commands (like setting volume to 0?) properly but if there are repeated requests it does respond to some of them. But this explanation sounds far fetched so it might not be the right one + this change was introduced in 4.0.1, in 4.0.2 nothing happened that could cause this problem.

Weird. :)

**Comment by ckirches** [comment]
I can confirm that mute does NOT work on Samsung C34J79. I can hear sound if the slider is at 0%.
This is in 4.0.2.

**Comment by waydabber** [comment]
@ckirches - do you have the same problem as @carefulcomputer - you can mute with 4.0.0 and not with 4.0.2 - or is it a generic display problem?

**Comment by ellieplz** [comment]
> I can confirm that mute does NOT work on Samsung C34J79. I can hear sound if the slider is at 0%.
> This is in 4.0.2.

Same exact monitor and same exact issue.

**Comment by waydabber** [comment]
I added a bug label to this one but I am not sure how to handle this. Display firmware implementation for mute varies a lot among displays and any way I change the code some number of users/displays will have issues. :/ We'll have to delve into this later, but did put a lot of time into trying to fix this one before, the current implementation seems to be compatible with most (but obviously not all) displays.

**Comment by ellieplz** [comment]
No worries at all! Your work is greatly appreciated, but do let us know if we can provide anymore details.

Would a switch, or drop down like "if mute feature/setting volume to 0 is not working press this, otherwise stay with default" work? did you have some build in the past that was working on Samsung C34J79x monitors?

**Comment by waydabber** [comment]
I am not sure if it did ever worked. Right now if `Enable Mute DDC command` is not enabled, the app will set the volume to 0, if the mute is enabled then it won't set to 0 (as that confuses some displays) but sends mute/umute commands instead. If neither works, then I don't really have an idea what to do. Maybe it could send 0 and use mute at the same time as a third option. You can try if this works if you have XCode, I can let you know where to change the code.

**Comment by ellieplz** [comment]
Oh! `"Enable Mute DDC command"`in advanced settings did work immediately, thank you so much!

**Comment by ghost** [comment]
My advanced Enable Mute DDC command is greyed out (everything is greyed out except the Brightness Available check mark) - any ideas on how to access it? Seems like it would be the fix needed for not able to mute on zero...

I'm on a Samsung C34J79x and Macbook Air M2

**Comment by waydabber** [comment]
@mwho1 - this might be because DDC is not available for the display for some reason.

**Comment by waydabber** [comment]
For mute related issues: https://github.com/waydabber/BetterDisplay/issues/1248
